driver/ppm_events_public.h

```diff
@@ -1302,7 +1302,16 @@ struct ppm_param_info {
 	const void *info; /**< If this is a flags parameter, it points to an array of ppm_name_value,
 			       else if this is a dynamic parameter it points to an array of ppm_param_info */
 	uint8_t ninfo; /**< Number of entry in the info array. */
-} __attribute__((packed));
+}
+#if defined(_WIN64) || defined(WIN64) || defined(_WIN32) || defined(WIN32)
+#pragma pack(push,1)
```

Comment:
We can't hide this in a `#define`, can we?

Reply:
I tried a few things but including defines within defines seems to not work in Visual Studio. Since the number of packed structs is not too large, can we live with this solution for now?

Reply:
hmm, how about:

```
#if defined(_WIN64) || ...
#define SYSDIG_WIN 1
#endif
```
or (if that fails, though I see no reason why it should)
```
#if defined(_WIN64)
#define SYSDIG_WIN 1
#elif defined(WIN64)
#define SYSDIG_WIN 1
... etc.
#endif
```

Also, there's a `__pragma()` directive we might be able to use instead of `#pragma`, maybe something like:

```
#ifdef SYSDIG_WIN
#define _packed __pragma(pack(push, 1)) ; __pragma(pack(pop))
#else
#define _packed __attribute__((packed))
#endif
```

and later:

```
struct { ... } _packed;
```

(there will be an extra ; in the windows case, not sure if that makes it invalid C or C++, or just an empty declaration)

https://msdn.microsoft.com/en-us/library/d9x1s805.aspx

Reply:
I wasn't aware of the __pragma directive. That did the trick!
BTW I can avoid the extra ; in windows by moving the ; into the _packed define in non-windows case: #define _packed __attribute__((packed));

Then the struct definition would be struct { ... } _packed
with no trailing ; 
The code looks a bit odd though. The extra ; in windows doesn't seem to be causing a compilation issue, so we can live with that. What do you think?